Source/JavaScriptCore/runtime/LiteralParser.cpp

```diff
@@ -869,18 +869,18 @@ ALWAYS_INLINE TokenType LiteralParser<CharType>::Lexer::lexString(LiteralParserT
         } else {
             ([&]() ALWAYS_INLINE_LAMBDA {
 #if CPU(ARM64) || CPU(X86_64)
-                constexpr size_t stride = 16 / sizeof(CharType);
+                constexpr size_t stride = SIMD::stride<CharType>;
```

Comment:
Other than the specifics of what characters are being searched for, this looks very close to same code as is used in the HTMLFastPath code below. 

Is there a common bit that could be pulled out? Something like the following (totally untested or compiled, just written here in GitHub).

```c++
template<typename CharType, CharType... Characters>
const CharType* searchFor(const CharType* ptr, const CharType* end)
{
    constexpr size_t stride = SIMD::stride<CharType>;
    using UnsignedType = std::make_unsigned_t<CharType>;
    if (static_cast<size_t>(end - ptr) >= stride) {
        auto match = [&](auto* cursor) ALWAYS_INLINE_LAMBDA {
            auto input = SIMD::load(bitwise_cast<const UnsignedType*>(cursor));
            auto mask = SIMD::bitOr(SIMD::equal(input, SIMD::splat<UnsignedType>(Characters))...);
            return SIMD::findFirstNonZeroIndex(mask);
        };

        for (; ptr + (stride - 1) < end; ptr += stride) {
            if (auto index = match(ptr)) {
                ptr += index.value();
                return ptr;
            }
        }
        if (ptr < end) {
            if (auto index = match(end - stride)) {
                ptr = end - stride + index.value();
                return ptr;
            }
            ptr = end;
        }
        return ptr;
    }

    for (; ptr != end; ++ptr) {
        auto character = *ptr;
        if (... || (character == characters))
            return ptr;
    }

    return ptr;
}
```

Reply:
Right now, this patch is focusing on adding optimization. Cleaning up will be done in a separate change.

Reply:
Makes sense.